src/webapi/node.rs

```diff
@@ -383,13 +383,27 @@ impl Node {
         js_try!(
             var span = document.createElement("span");
             span.innerHTML = @{html};
-            if( span.childNodes.length != 1 ) {
+
+            // Fast path, if we have one childNode just return it.
+            if( span.childNodes.length > 1 ) {
+                // We will drop one leading and training whitespace-only text node.
+                if( span.firstChild.nodeType == Node.TEXT_NODE &&
+                    !span.firstChild.textContent.trim() ) {
+                    span.removeChild(span.firstChild);
+                }
+                if( span.lastChild.nodeType == Node.TEXT_NODE &&
+                    !span.lastChild.textContent.trim() ) {
+                    span.removeChild(span.lastChild);
+                }
+            }
+
+            if( span.childNodes.length != 1) {
```

Comment:
Hm... all of this seems unnecessary. I mean - wouldn't replacing `@{html}` with `@{html.trim()}` achieve the same thing? (:

Reply:
I originally used `.trim()` in rust but this changes the behavior for people generating text nodes. For example

```rust
stdweb::web::Node::from_html(r#"
  foobar
"#);
```

Currently this has a text node surrounded by whitespace but if we used trim it would just be the word. The checks here ensure that I am only changing behavior which was previously an error.

Reply:
Hmm.. in which case can you add a test for the text-surrounded-by-whitespace case?

I think we still may be able to do this on Rust's side (which should be faster) - find first non-whitespace byte, if it's `b'<'` then it's not a text node and should be trimmed (and since we've found the first non-whitespace byte we already have it trimmed from the left side), otherwise do not touch it. Also, Pauan is right in that Rust's `trim` is Unicode-aware, which we don't really need here; only `b'\n'`, `b'\r'`, `b'\t'` and `b' '` should suffice.

Reply:
@koute 

I'll try to implement this rust side.

> Rust's trim is Unicode-aware

Is this a problem? Should I use a trim which explicitly isn't unicode aware?

@Pauan I could be convinced not to however there are two reasons I didn't do this.

1. This is a change in behaviour to existing users.
1. Sometimes you do need to have space around text as it will affect the DOM and rendering.

If we are okay with these points adding `.trim()` is trivial.

Reply:
> Is this a problem? Should I use a trim which explicitly isn't unicode aware?

It's not strictly a problem, but it's totally unnecessary.

I'm generally pretty ambivalent as to which behavior we pick here, however since this method's *current* behavior is that it leaves text nodes as-is we shouldn't change that without a major version bump, so I'm okay with trimming only actual non-text nodes here provided we'll do it on Rust's side with a custom non-Unicode trimmer (which should be fairly easy to write).

Reply:
@kevincox Just to be clear, I never meant for stdweb to automatically call `trim`, instead I meant that the *user* of stdweb would call `trim`:

```rust
Node::from_html(foo.trim())
```

This allows the user to call `trim` only when it makes sense to do so.